microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -1,113 +1,139 @@
+* **Back-end code**: the endpoints needed by your app, when connected to the internet, to retrieve dynamic content that may be stored in a database on your server.
+* **Front-end code**: the resources needed for the app to be installed on the user's device, such as HTML, CSS, and JavaScript code.
 
-1. Create a new file in the project by pressing `Ctrl+N`, add the following content, and save the file as `index.html`:
+Your back-end code can use the server-side languages of your choice such as ASP.NET, Java, Node.js, or PHP. Note, however, that server-side endpoints may not even be required depending on the app your're building. The PWA that you create in this tutorial doesn't have any server-side code, because the app exclusively runs on the device it's installed on, and doesn't need any server-side data.
 
-   ```html
-   <!DOCTYPE html>
-   <html lang="en-US" dir="ltr">
-     <head>
-       <meta charset="UTF-8" />
-       <meta name="viewport" content="width=device-width,initial-scale=1" />
-       <link rel="shortcut icon" href="https://c.s-microsoft.com/favicon.ico?v2" />
-       <title>My Sample PWA</title>
-     </head>
-     <body>
-       <h1>My Sample PWA</h1>
-     </body>
-   </html>
-   ```
+Your front-end code uses HTML, CSS, JavaScript, and a JSON manifest only.
 
-1. Start a local web server by using the `http-server` Node.js library:
+You use HTML to describe the content in your app, such as the text, images, text fields, or buttons that appear in the user interface. You then use CSS to organize the HTML content in a layout, and provide styles to elements. You use JavaScript to add user interactions to your user interface. And finally, you use a JSON manifest file that describes your application to the host operating system.
 
-   ```shell
-   cd path\to\MySamplePWA
-   npx http-server
-   ```
-  
-   In the above commands, replace `path\to\MySamplePWA` with the actual path where you create the `MySamplePWA` folder.
+Note that although your front-end code runs by using the device's web browser, the browser user interface may not be visible as your app can choose to run in a standalone window.
 
-You now have a simple, functional website running on a local web server.
+On top of the user interface code, you also use JavaScript to make your application faster, more reliable, and network-independent by using a service worker file. Finally, your front-end code also contains a JSON manifest file that describes your application to the host operating system.
 
-Now browse to `http://localhost:8080` to view your new website.
+Below is a diagram of the high-level architecture of a PWA:
 
-![Running your new PWA on localhost.](../media/sample-pwa-app.png)
+![Architecture diagram of a PWA, showing the web server on one side, and the device on the other. The device part contains the front-end code with HTML, CSS, JavaScript, the service worker, and the manifest](../media/pwa-architecture.png)
```

Comment:
```suggestion
![Architecture diagram of a PWA](../media/pwa-architecture.png)
```
pulled out length alt-text as regular text that's visible to all users.

internal version of DCG gives a suggested(?) char limit for alt-text. It's short, like 35(?) chars.
Public version:
https://learn.microsoft.com/en-us/contribute/markdown-reference#standard-conceptual-images-default-markdown
https://www.bing.com/search?q=html+length+of+alt+text